### PR TITLE
Number template

### PR DESCRIPTION
**Pull request by cdunn2001**
See #423 and #424.

### CODE REVIEW

File: include/json/thing.h [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 /** \brief JSON (JavaScript Object Notation).
  */
+template<Integer, Float>
```
Comment (BillyDonahue) [inline include/json/thing.h]: I don't get it. Why is template<Integer,Float> above a namespace?

**Comment by cdunn2001** [comment]
Oops. Updated.

**Comment by cdunn2001** [comment]
Any comments?

Basically, we would name the template something obvious, then typedef `Json::Value<std::string, int, double>` to it. Users would have to change a few number-conversion function calls, but that would probably improve their code.

If anyone wants their own datatypes, they must provide `toJson()` and `fromJson()` converter specializations. We would need to settle on their signatures. `toJson()` is the hardest because it involves memory allocation.

**Comment by okodron** [comment]
Is it still relevant?

**Comment by cdunn2001** [comment]
It's future planning. In my opinion, this is closer to how jsoncpp parsers should look. The user should choose both the internal datatypes and how they are converted to/from JSON data. (We could provide useful defaults.) Not only is that flexible, but it also would eliminate a large fraction of bug reports (here and elsewhere).

**Pull request closed without merging**